### DISCUSSION TITLE
Standalone instances

### DISCUSSION THREAD

**Discussion by hoekma** [Grokzen/docker-redis-cluster#108] [Q&A]
Out of curiosity, why do you have multiple standalone instances and why put them on ports 7006 and 7007?  Don't most people who will use a standalone instance would just want one instance on 6379.

I have a local version that works with the rest of your setup that just puts one on 6379 if REDIS_USE_STANDALONE=true.  I needed it to make it easier to test standalone and cluster setups in my RedPop project.  Happy to share a PR if you're interested.

**Comment by Grokzen** [comment]
The answer to this is that this project and docker container came out from testing when i develop this lib https://github.com/Grokzen/redis-py-cluster where we need both a clustered instance and a few standalone instances and it was just most easy to provide both of them in a single container and it is something that has lived on over time and if you are in the spot where you want a standalone instance only i would not use this docker image and i would just use the default base upstream redis docker image which gives you a normal single instance on port 6379.
Right now i will not look into changing the default port back to 6379 because if you want that you can just use the default redis image, but i would approve a PR where you make the port variable where you can change it to a new starting port range like we already have for the cluster with the environment variable `INITIAL_PORT` and make one for the standalone version.

**Reply by hoekma** [reply]
Ok, I'll take a look at that -- to be sure I understand your proposal, the standalone instances would be able to have an independent, contiguous range separate from the cluster's contiguous port range.

**Reply by hoekma** [reply]
Oh, also, I don't know if you saw my comment on the passwords issue: https://github.com/Grokzen/docker-redis-cluster/issues/54

I do have a solution that I could provide a PR for that allows optional passwords to be set for default and a custom user by assigning environment variables on the machine that the MAKE BUILD is run on...such as : 

REDIS_CLUSTER_IP=0.0.0.0
REDIS_DEFAULT_PASSWORD=
REDIS_USER_NAME=hoek
REDIS_USER_PASSWORD=hoekpass
REDIS_USE_STANDALONE=true  

In the above example, the default password is not set but a redis user will be created.  If the default password is set then it will require a password to run commands.  If no REDIS_USER_NAME then no user will be created.   It's backwards compatible with your existing implementation and an independent PR from the one we've been discussing about the standalone. Happy to share.

**Reply by Grokzen** [reply]
Try to make a PR, it is easier to see the solution fully complete to review it then compared to review it like this in theory. As long as the current default setup is not touch but you add in parameters to configure the container i am happy.

**Reply by hoekma** [reply]
Get this when I'm trying to push: 
```
ERROR: Permission to grokzen/docker-redis-cluster.git denied to hoekma.
fatal: Could not read from remote repository.
```

**Reply by Grokzen** [reply]
@hoekma Well that is not so strange as it looks like you are trying to push to my repo directly and not to your own fork of this repo. Try to fork this and clone from there and you should have much more success.

**Reply by hoekma** [reply]
Ah ok... sorry, been working in private repos for too long now I guess.  PR is up from my fork.
